react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -212,12 +212,18 @@ const analyzeNodesAgainstExpression = ({
         );
 
         // get the 3d nodes linked to the asset and with treeindex and subtreeRange
-        const treeNodes: NodeAndRange[] = await getThreeDNodesFromAsset(nodesFromThisAsset, model);
+        const nodesAndRange: NodeAndRange[] = await getThreeDNodesFromAsset(
+          nodesFromThisAsset,
+          model
+        );
 
-        applyNodeStyles(treeNodes, outputSelected, model);
+        return nodesAndRange;
       }
     })
   );
+
+  const filteredAllTreeNodes = allTreeNodes.flat().filter((node) => node !== undefined);
```

Comment:
When I look at it again now, I also don't understand how any element in `allTreeNodes` becomes undefined :thinking: Does Typescript give you that type? Does it have to do with the Promise somehow? :thinking: